### PR TITLE
feat: improve connectors flow

### PR DESCRIPTION
**Pull request by luizstacio**
## UI Components

- Improve dialog flows for, connect, disclaimer, install, bridge and switch network.
- Refactor code around to ensure consistency 

## Hooks and Connectors

- Add `query` options for customization of `useNamedQuery` on hooks. This can help to improve number of requests in cases of multiple checks like balances.

    ```ts
    import { useBalance } from '@fuels/react';
    const { balance } = useBalance({ address: '0x1234', assetId: '0x1234', query: { refetchInterval: 1000 } });
    ```

- Add Networks configuration to `FuelUIProvider` to allow users to provide a list of supported networks with their repesctive bridges.
- Add `suggestBridge` option to `FuelUIProvider` to enable/disable the bridge dialog for users flows.

    ```ts
    <FuelProvider
    uiConfig={{
        suggestBridge: true,
        networks: [
        {
            chainId: CHAIN_IDS.fuel.testnet,
            bridgeURL: 'https://app.fuel.network/bridge?from=eth&to=fuel&auto_close=true',
        },
        ],
    }}>
    </FuelProvider>
    ```

- Add `useIsSupportedNetwork` hook to check if the current network is supported by the user.

    ```ts
    import { useIsSupportedNetwork } from '@fuels/react';
    const { isSupportedNetwork } = useIsSupportedNetwork();
    ```
- Improve predicates setupProviders to remove delays by adding a new controlled flag `hasProviderSuccessed`.

### CODE REVIEW

**Review by helciofranco — DISMISSED** [review]
awesome

File: packages/react/src/ui/Connect/index.tsx [hunk 1]
```diff
@@ -1,108 +1,61 @@
...
   Divider,
-  FuelRoot,
 } from './styles';
 
-import type { FuelConnector } from 'fuels';
-import { getThemeVariables } from '../../constants/themes';
-import { DialogContent } from '../Dialog/components/Content';
-import { Bridge } from './components/Bridge/Bridge';
 import { Connecting } from './components/Connector/Connecting';
+// import { Bridge } from './components/Bridge/BridgeDialog';
```
Comment (arthurgeron) [inline packages/react/src/ui/Connect/index.tsx:15]:
```suggestion
```

File: packages/react/src/hooks/useCurrentConnector.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}: UseCurrentConnectorParams<TName, TData> = {}) => {
+  const { fuel } = useFuel();
+  return useNamedQuery('connector', {
+    queryKey: QUERY_KEYS.currentConnector(),
+    queryFn: async () => {
+      const isConnected = await fuel.isConnected();
+      if (!isConnected) return null;
+      console.log('currentConnector', fuel.currentConnector());
+      const connector = fuel.currentConnector() ?? null;
+      return connector;
```
Comment (arthurgeron) [inline packages/react/src/hooks/useCurrentConnector.tsx:45]:
```suggestion
      return fuel.currentConnector() ?? null;
```

**Review by arthurgeron — CHANGES_REQUESTED** [review]
Good stuff. Just few things to take a look at.

**Comment by arthurgeron** [comment]
Should it look like this on testnet?
CleanShot 2024-09-25 at 20 48 20

**Comment by luizstacio** [comment]
> Should it look like this on testnet? CleanShot 2024-09-25 at 20 48 20

Done, I improve this part in the way other DApps should also do, thanks for the comment.

**Review by arthurgeron — APPROVED** [review]

File: packages/react/src/providers/FuelUIProvider.tsx [hunk 3]
```diff
@@ -13,41 +13,45 @@ import { useConnect } from '../hooks/useConnect';
...
+  }) => void;
   setError: (error: Error | null) => void;
   error: Error | null;
   dialog: {
     connector: FuelConnector | null;
     isOpen: boolean;
     back: () => void;
     connect: (connector: FuelConnector) => void;
-    retryConnect: () => Promise<void>;
+    retryConnect: (connector: FuelConnector) => Promise<void>;
```
Comment (LuizAsFight) [inline packages/react/src/providers/FuelUIProvider.tsx:54]: make it optional to avoid breaking change?

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Comment by LuizAsFight** [comment]
add `margin-top: 4px;` to this el to fix spacing

image

**Comment by LuizAsFight** [comment]
after connecting to a account, then disconnection, then connecting to another account but without funds, the bridge popup didnt show eu

File: packages/react/src/providers/FuelUIProvider.tsx (hunk 3 above)
Comment (luizstacio) [reply packages/react/src/providers/FuelUIProvider.tsx:54]: This is new is not released yet.

File: packages/react/src/providers/FuelProvider.tsx [hunk 4]
```diff
@@ -2,51 +2,59 @@ import type { FuelConfig } from 'fuels';
...
+  networks,
 }: FuelProviderProps) {
   const theme = _theme || 'light';
+  const networksConfig = useNetworkConfigs(networks);
+  const uiConfig = Object.assign(
+    {
+      suggestBridge: true,
+    },
+    _uiConfig ?? {},
+  );
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/react/src/providers/FuelProvider.tsx:37]: I can change the default to be false, but I would assign a object anyway, as this is the safer way to ensure code is consistent.

File: packages/react/src/hooks/useWallet.ts [hunk 5]
```diff
@@ -18,21 +35,37 @@ import { QUERY_KEYS } from '../utils';
...
  * ```
  * To get the current account's wallet:
  * ```ts
  * const { wallet } = useWallet();
  * ```
  */
-export const useWallet = (address?: string | null) => {
+export function useWallet(
+  params?: UseWalletParams,
+): DefinedNamedUseQueryResult<'wallet', Account | null, Error>;
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/react/src/hooks/useWallet.ts:47]: It's not a break change is deprecation, that keep supporting the old way.

File: packages/docs/src/guide/react-hooks/useCurrentConnector.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+To fetch connectors:
+```ts
+const { connectors } = useConnectors();
+console.log(connectors);
+```
+
+#### Defined in
+[packages/react/src/hooks/useCurrentConnector.tsx:31](https://github.com/fuellabs/fuel-connectors/blob/main/packages/react/src/hooks/useCurrentConnector.tsx#L31)
+
+___
```
[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/docs/src/guide/react-hooks/useCurrentConnector.md:28]: This is auto generated, maybe I need to update comments somewhere

**Comment by luizstacio** [comment]
> after connecting to a account, then disconnection, then connecting to another account but without funds, the bridge popup didnt show eu

This is expected, to avoid creep control, adding to change change the dismiss based on account can lead to blinking issues, I think is a good think to create another issue.

**Review by LuizAsFight — APPROVED** [review]

**Pull request merged**